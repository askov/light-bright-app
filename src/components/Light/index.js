import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function Light(props) {
  return (
    <div className="light"
         onMouseEnter={() => props.handleLightEnter(props.index)}
         onMouseDown={() => props.handleLightClick(props.index)}
         onDoubleClick={() => props.handleDoubleClick(props.index)}
         style={{backgroundColor: props.color}}>
    </div>
  );
}

Light.propTypes = {
  index: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  handleLightEnter: PropTypes.func.isRequired,
  handleLightClick: PropTypes.func.isRequired,
  handleDoubleClick: PropTypes.func.isRequired,
};

export default Light;
