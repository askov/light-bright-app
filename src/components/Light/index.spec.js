import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Light from './index';

describe('<Light />', () => {
  let props;
  let wrapper;

  const createWrapper = (props) => {
    return shallow(<Light {...props} />);
  }

  beforeEach(() => {
    props = {
      index: 42,
      color: 'red',
      handleLightEnter: jest.fn(),
      handleLightClick: jest.fn(),
      handleDoubleClick: jest.fn(),
    };
    wrapper = createWrapper(props);
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(<Light {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders light container div', () => {
    expect(wrapper.find('div.light').length).toBe(1);
  });

  test('div style background color is set from props', () => {
    const p = wrapper.find('div.light').props();
    expect(p).toHaveProperty('style.backgroundColor', props.color);
  });

  test('handles mouse enter event properly', () => {
    wrapper.simulate('mouseenter');
    expect(props.handleLightEnter.mock.calls.length).toBe(1);
    expect(props.handleLightEnter).toBeCalledWith(props.index);
  });

  test('handles mouse down event properly', () => {
    wrapper.simulate('mousedown');
    expect(props.handleLightClick.mock.calls.length).toBe(1);
    expect(props.handleLightClick).toBeCalledWith(props.index);
  });

  test('handles double click event properly', () => {
    wrapper.simulate('dblclick');
    expect(props.handleDoubleClick.mock.calls.length).toBe(1);
    expect(props.handleDoubleClick).toBeCalledWith(props.index);
  });
});