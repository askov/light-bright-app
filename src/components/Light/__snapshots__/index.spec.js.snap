// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Light /> renders correctly 1`] = `
<div
  className="light"
  onDoubleClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;
